airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionHandler.kt

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+package io.airbyte.cdk.integrations.util
+
+import io.airbyte.cdk.integrations.base.AirbyteTraceMessageUtility
+import io.airbyte.cdk.integrations.base.Command
+import io.airbyte.cdk.integrations.base.errors.messages.ErrorMessage
+import io.airbyte.commons.exceptions.ConfigErrorException
+import io.airbyte.commons.exceptions.ConnectionErrorException
+import io.airbyte.commons.exceptions.TransientErrorException
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.util.function.Consumer
+import kotlin.system.exitProcess
+import org.jetbrains.annotations.VisibleForTesting
+
+private val LOGGER = KotlinLogging.logger {}
+
+const val COMMON_EXCEPTION_MESSAGE_TEMPLATE: String =
+    "Could not connect with provided configuration. Error: %s"
+
+const val DATABASE_CONNECTION_ERROR: String =
+    "Encountered an error while connecting to the database error"
+
+enum class FailureType {
+    CONFIG,
+    TRANSIENT
+}
+
+data class ConnectorErrorProfile(
+    var errorClass: String,
```

Comment:
perhaps we should have some kind of dictionary or registry for these kinds of values? would it make sense to strongly type these values? like an enum value or some union type? then again perhaps free-form strings are fine too, just thinking aloud

Reply:
I think in the future, maybe... currently we don't have an immediate use case for this field.. so I feel we could improve it in the future. Maybe I shall remove this field as it was not really used perhaps...?

Reply:
In the absence of a clear vision, let's leave things as they are. I think it's OK to want to group things into classes of errors. We can always do this kind of taxonomy exercise later on. I have a dream where Airbyte has meaningful error codes.